packages/react-bootstrap-table2/test/row-selection/selection-header-cell.test.js

```diff
@@ -0,0 +1,137 @@
+    describe('when props.mode is radio', () => {
+      it('should not update component', () => {
+        wrapper = shallow(<SelectionHeaderCell mode={'radio'} />);
+
+        expect(wrapper.instance().shouldComponentUpdate({})).toBe(false);
+      });
+    });
+
+    describe('when props.mode is checkbox', () => {
+      describe('if checkedStatus prop has not been changed', () => {
+        it('should not update component', () => {
+          const checkedStatus = Constant.CHECKBOX_STATUS_CHECKED;
+          const nextProps = { checkedStatus };
+
+          wrapper = shallow(
+            <SelectionHeaderCell mode={'checkbox'} checkedStatus={checkedStatus} />);
+
+          expect(wrapper.instance().shouldComponentUpdate(nextProps)).toBe(false);
+        });
+      });
+
+      describe('if checkedStatus prop has been changed', () => {
+        it('should update component', () => {
+          const { CHECKBOX_STATUS_INDETERMINATE, CHECKBOX_STATUS_CHECKED } = Constant;
+          const checkedStatus = CHECKBOX_STATUS_CHECKED;
+          const nextProps = { checkedStatus };
+
+          wrapper = shallow(
+            <SelectionHeaderCell mode={'checkbox'} checkedStatus={CHECKBOX_STATUS_INDETERMINATE} />);
+
+          expect(wrapper.instance().shouldComponentUpdate(nextProps)).toBe(true);
+        });
+      });
+    });
+  });
+
+  describe('handleCheckBoxClick', () => {
+    describe('when th was clicked', () => {
+      let spy;
+      const mockOnAllRowsSelect = jest.fn();
+
+      describe('if props.mode is radio', () => {
+        it('should do nothing', () => {
+          spy = jest.spyOn(SelectionHeaderCell.prototype, 'handleCheckBoxClick');
+
+          wrapper = shallow(
+            <SelectionHeaderCell
+              mode={'radio'}
+              checkedStatus={Constant.CHECKBOX_STATUS_CHECKED}
+              onAllRowsSelect={mockOnAllRowsSelect}
+            />);
+          wrapper.find('th').simulate('click');
+
+          expect(spy).not.toHaveBeenCalled();
+          expect(mockOnAllRowsSelect).not.toHaveBeenCalled();
+        });
+      });
+
+      describe('if props.mode is checkbox', () => {
+        it('should call handleCheckBoxClick', () => {
+          spy = jest.spyOn(SelectionHeaderCell.prototype, 'handleCheckBoxClick');
+
+          wrapper = shallow(
+            <SelectionHeaderCell
+              mode={'checkbox'}
+              checkedStatus={Constant.CHECKBOX_STATUS_CHECKED}
+              onAllRowsSelect={mockOnAllRowsSelect}
+            />);
+
+          wrapper.find('th').simulate('click');
+
+          expect(spy).toHaveBeenCalled();
+          expect(mockOnAllRowsSelect).toHaveBeenCalled();
+        });
+      });
+    });
+  });
+
+  describe('render', () => {
+    describe('when props.mode is radio', () => {
+      beforeEach(() => {
+        const checkedStatus = Constant.CHECKBOX_STATUS_CHECKED;
+
+        wrapper = shallow(<SelectionHeaderCell mode={'radio'} checkedStatus={checkedStatus} />);
```

Comment:
`mode={'radio'}` can be `mode="radio" `, please change following `mode={'checkbox'} `